sponsorship_compassion/models/res_partner.py

```diff
@@ -398,6 +398,26 @@ def _random_str():
         # referenced users are unlinked to avoid error when self.active is set to False
         self.user_ids.sudo().unlink()
 
+        # Delete message and mail history
+        self.message_ids.unlink()
+        self.env["mail.mail"].search([("recipient_ids", "=", self.id)]).unlink()
+        self.env["ir.attachment"].search(
+            [("res_model", "=", "res.partner"), ("res_id", "=", self.id)]
+        ).unlink()
+        self.bank_ids.unlink()
+        self.privacy_statement_ids.unlink()
+        self.env["gmc.message"].search([("partner_id", "=", self.id)]).write(
+            {"res_name": self.name}
+        )
+
+        # Delete all logs
+        # Use a subquery to get the mail_message IDs associated with self.id
+        mail_message_ids = self.env['mail.message'].search([('res_id', '=', self.id)]).ids
+        # Delete mail_tracking_value records where mail_message_id matches the IDs from the subquery
+        self.env['mail.tracking.value'].search([('mail_message_id', 'in', mail_message_ids)]).unlink()
+        # Now we are sure that all mail_message_ids have been deleted, we can delete the mail_message records
+        self.env['mail.message'].search([('res_id', '=', self.id)]).unlink()
```

Comment:
I would say that all this part is redundant to what is done in line 402 and should be removed. Can you try without it?